File: backend/Models/Entities/Leave.cs
namespace backend.Models.Entities
{
    public class Leave
    {
        public int Id { get; set; }

        public string EmployeeId { get; set; } = null!;

        public string LeaveType { get; set; } = null!;

        public DateTime StartDate { get; set; }

        public DateTime EndDate { get; set; }

        public string? Reason { get; set; }

        public string Status { get; set; } = "Pending";

        public string? ApprovedById { get; set; }

        public DateTime AppliedOn { get; set; } = DateTime.UtcNow;

        public Employee Employee { get; set; } = null!;

        public Employee? ApprovedBy { get; set; }
    }
}

File: frontend/src/api/models/leave.ts
export interface CreateLeaveRequest {
  leaveType: string;
  startDate: string;
  endDate: string;
  reason: string;
}

export interface UpdateLeaveRequest {
  leaveType: string;
  startDate: string;
  endDate: string;
  reason: string;
}

export interface Leave {
  id: number;
  employeeId: string;
  employeeName: string;
  leaveType: string;
  startDate: string;
  endDate: string;
  reason: string;
  status: string;
  approvedById: string;
  approvedByName: string;
  appliedOn: string;
}

export interface LeaveQueryParams {
  PageNumber?: number;
  PageSize?: number;
  employeeId?: string;
  status?: string;
  leaveType?: string;
  startDate?: string;
  endDate?: string;
}